profiles/bicycle.lua

```diff
@@ -367,35 +368,73 @@ function handle_bicycle_tags(profile,way,result,data)
   end
 
   -- cycleway
-  local has_cycleway_left = false
-  local has_cycleway_right = false
+  local has_cycleway_forward = false
+  local has_cycleway_backward = false
+
+  -- cycleways on normal roads
+  if result.forward_mode ~= mode.inaccessible and result.backward_mode ~= mode.inaccessible and not implied_oneway then
+
+    if cycleway and profile.cycleway_tags[cycleway] and string.find(cycleway, "opposite") ~= 1 then
+
+      has_cycleway_backward = true
+      has_cycleway_forward = true
+
+    end
+    if (cycleway_right and profile.cycleway_tags[cycleway_right] and string.find(cycleway_right, "opposite") ~= 1)
+       or (cycleway_left and profile.cycleway_tags[cycleway_left] and string.find(cycleway_left, "opposite") == 1) then
```

Comment:
Maybe we can simplify this logic a little bit by creating a helper function that handles bi-directional tag lookup like that, similar to `get_forward_backward_by_key` in `tags.lua`.